avocado/core/dispatcher.py

```diff
@@ -59,3 +59,26 @@ class CLICmdDispatcher(Dispatcher):
 
     def __init__(self):
         super(CLICmdDispatcher, self).__init__('avocado.plugins.cli.cmd')
+
+
+class JobPrePostDispatcher(Dispatcher):
+
+    """
+    Calls extensions before Job execution
+
+    Automatically adds all the extension with entry points registered under
+    'avocado.plugins.job.prepost'
+    """
+
+    def __init__(self):
+        super(JobPrePostDispatcher, self).__init__('avocado.plugins.job.prepost')
+
+    def map_methods(self, method_name, job):
+        for ext in self.extensions:
+            try:
+                if hasattr(ext.obj, method_name):
+                    method = getattr(ext.obj, method_name)
+                    method(job)
+            except Exception as e:
+                job.log.error('Error running method "%s" of plugin "%s": %s',
+                              method_name, ext.name, e)
```

Comment:
You mean:

```
try:
   ...
except:
   ...
```

Right? IIRC I chose `Exception` because it would not catch exceptions such as `KeyboardInterrupt`, which seemed appropriate... BTW, what do you mean by "incorrect" exceptions?